### PR TITLE
Run webkit tests on pipeline builds

### LINKED ISSUE #990: Improve cross browser test coverage: enable WebKit tests on CI
## 🧹 Tech Debt

We currently run unit tests in only Chromium and our Chromatic tests in Chromium and Firefox. More test coverage would be better to help catch issues like #984 via automated rather than manual testing.

Possible directions (might split this issue to track these separately)

1. ~~add commands to run unit tests on other browsers and run them locally (I tried this briefly and discovered several failing in Safari and Firefox)~~ **Done**

2. run tests as part of our PR build. Would require some research to find out best practices for getting agents with Firefox and webkit **Done for Firefox**
   - mac agents? Probably not as they cost many times more than default Linux agents
   - Using Playwright's cross-platform webkit build? Related: #705
   - ~~we'd also want a way to disable specific tests in specific browsers. We could consider the approach we use for WebVIs where certain tags are added to test names.~~ **Done**

3. Consider whether we want to run cross-browser testing for nimble-tokens, nimble-angular, and nimble-blazor

Already done by @fredvisser :
- Once it's available, enable Chromatic testing in Safari (at a recent meeting we heard it was coming in February)

```[tasklist]
## Related PRs
- [ ] #998 
- [ ] #1067 
```

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Closes #990

## 👩‍💻 Implementation

I [upstreamed a change](https://github.com/google/karma-webkit-launcher/pull/13) to `karma-webkit-launcher` to fix our issues on Linux build machines, and now we're pulling in the fixed release.

## 🧪 Testing

Build runs without hanging. The validation step still takes roughly 6 minutes, even though it is running most of the `nimble-components` tests an additional time.

```
[test-concurrent:nimble-components] [test-webkit:verbose] Safari 17.0 (Linux x86_64): Executed 4595 of 4646 (skipped 51) SUCCESS (5 mins 11.218 secs / 4 mins 17.806 secs)
```

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/karma-webkit-launcher-nimble/README.md
Comment (rajsite) [inline packages/karma-webkit-launcher-nimble/README.md:5]: Is this something we can upstream? I'm not particularly excited to fork and maintain all this

Comment (rajsite) [reply packages/karma-webkit-launcher-nimble/README.md:5]: their docs talk alot about playwright so I suspect they are interested in it working well

Comment (m-akinc) [reply packages/karma-webkit-launcher-nimble/README.md:5]: I wonder if they'll be satisfied with a purely empirical justification, or if they'll want to better understand why this is happening. I can see what they say, though. On the other hand, that repo has had 38 total commits in the 3 years of its history, so I bet we'd be fine never touching these three files again.

Comment (rajsite) [reply packages/karma-webkit-launcher-nimble/README.md:5]: yea thinks it's worth creating an issue discussing the problem in detail and opening an example PR upstream.

Comment (rajsite) [reply packages/karma-webkit-launcher-nimble/README.md:5]: Dunno the details but similar sounding issues seem to hit them before https://github.com/google/karma-webkit-launcher/issues/5

Comment (m-akinc) [reply packages/karma-webkit-launcher-nimble/README.md:5]: Created [issue](https://github.com/google/karma-webkit-launcher/issues/12) and [PR](https://github.com/google/karma-webkit-launcher/pull/13)

Comment (rajsite) [reply packages/karma-webkit-launcher-nimble/README.md:5]: FYI JIC looks like the PR is blocked on signing the Google CLA: https://github.com/google/karma-webkit-launcher/pull/13#issuecomment-2207649818

Comment (rajsite) [reply packages/karma-webkit-launcher-nimble/README.md:5]: May be worth pinging the person who self-assigned the issue that the CLA has been signed on the PR and it's ready to review. But even before that, just to confirm, did you test their workflows in your own fork? I don't see any actions run: https://github.com/m-akinc/karma-webkit-launcher/actions
Can help shorten review cycles to make sure their tests / workflows will pass.

Comment (m-akinc) [reply packages/karma-webkit-launcher-nimble/README.md:5]: I've now run the workflows, and they [all pass](https://github.com/m-akinc/karma-webkit-launcher/actions) except the Safari one, which they've [disabled on their repo](https://github.com/google/karma-webkit-launcher/actions/workflows/safari_tests.yml) (presumably because it's broken). I'll ping the guy.

Comment (m-akinc) [reply packages/karma-webkit-launcher-nimble/README.md:5]: Wow, it was merged immediately without feedback or questions and already released! I'll drop the forked copy and update our dependency.

File: packages/nimble-components/package.json
Comment (rajsite) [inline packages/nimble-components/package.json:47]: Only request since we don't do it on CI.
@m-akinc can you sanity check this running locally on windows?
@jattasNI can you sanity check this running locally on mac?
If all good then lets go!

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/package.json
Comment (m-akinc) [reply packages/nimble-components/package.json:47]: Works on windows

Comment (jattasNI) [reply packages/nimble-components/package.json:47]: Sorry I didn't get a chance to test this before vacation. I'll prioritize it next week.

Comment (rajsite) [reply packages/nimble-components/package.json:47]: Goal of asking for the mac audit was primarily for Jesse's benefit as our main mac dev. So I think we can go ahead and merge and Jesse can try it out when has a chance.

**Comment by m-akinc** [comment]
No real need for review from @DStavilaNI and @munteannatan -- just skipping two more wafer map tests on Webkit (I updated #2169 with the test names).

File: packages/nimble-components/package.json
Comment (jattasNI) [reply packages/nimble-components/package.json:47]: Confirmed it is ok on mac. There's a known (to me) issue where many/all of the `#SkipWebkit` tests fail in Firefox but that's pre-existing and I haven't prioritized finding a good way to disable them just on macOS.

**Pull request merged**
